plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -228,6 +228,40 @@ isProbablyIntegerEq (GHC.getName -> n)
       True
 isProbablyIntegerEq _ = False
 
+-- | Check for literal ranges like [1..9] and [1, 5..101].  This will also
+-- return `True` if there's an explicit use of `enumFromTo` or similar.
+isProbablyBoundedRange :: GHC.Id -> Bool
+isProbablyBoundedRange (GHC.getName -> n)
+    | Just m <- GHC.nameModule_maybe n
+    , GHC.moduleNameString (GHC.moduleName m) == "GHC.Enum" =
+        ("$fEnum" `isPrefixOf` methodName &&
+         (  "_$cenumFromTo"     `isSuffixOf` methodName  -- [1..100]
```

Comment:
Is there any better way of getting names of particular instances of methods like `enumFromTo`?  This was the best I could do and I'm not sure how robust it will be.  I suppose that if the names that GHC produces change then the tests will start to fail.

Reply:
Oh, this is too bad. I hadn't realized that GHC might specialize `enumFromTo`.

The answer to your question is, unfortunately, no. There's no way to prevent GHC from specializing a method. It is only possible to disable user written rules, not builtin rules.

Reply:
I'm not sure if we should match on the specialized name. I don't know if we can guarantee that that isn't an arbitrarily messed up version :thinking:

Reply:
We pass some information about GHC names through to the compiler in `NameInfo` in the `CompileContext`. Then you can use `getThing` to get the `TypedThing` and `getName` on that to get the `Name`. See how the stuff in `Builitns` works.

Reply:
> We pass some information about GHC names through to the compiler in `NameInfo` in the `CompileContext`. Then you can use `getThing` to get the `TypedThing` and `getName` on that to get the `Name`. See how the stuff in `Builitns` works.

Maybe we can come back to this in a later PR. I've run out  of  JIRA points on this.

Reply:
> I'm not sure if we should match on the specialized name. I don't know if we can guarantee that that isn't an arbitrarily messed up version 🤔

That's what I was asking about in my original comment.  Can we actaully get at the original name, especially after the inlining/rewriting?

Reply:
No, `NameInfo` is just for getting an actual GHC `Name` instead of string matching. I'm not sure it will match up if it has in fact been specialized. I'm not sure what we do in other cases, perhaps we just fail to spot stuff :thinking:

Reply:
I guess it's not too bad since we're failing instead of silently compiling differently.

Reply:
I guess since this is inherently heuristic rather than matching a precise name there's no point doing the `NameInfo` stuff and this is fine.